### DISCUSSION TITLE
Simulation fails at high current inputs

### DISCUSSION THREAD

**Discussion by shant4** [pybamm-team/PyBaMM#1632] [Q&A] [answered]
**Describe the bug**
I used the Chen2020 chemistry and lithium_ion.DFN model. When I tried to change the current function to 25 A, (5C_rate), it took more than 30 mins to run and the output voltage curve was sharp "V". If I change the model to SPM, the simulation would fail to run. Based on the tutorial, I expected to see the voltage curve ends earlier than the simulated time frame. But it failed to have results if the current is high. Is there a range for the current? 

**To Reproduce**
Steps to reproduce the behaviour:
1. change parameter_values["Current function [A]"] = 25
2. run

**Expected behaviour**
Terminal outputs for DFN model:
At t = 0.0453447, , mxstep steps taken before reaching tout.
At t = 0.0453447, , mxstep steps taken before reaching tout.
At t = 0.0453447, , mxstep steps taken before reaching tout.
Maximum number of decreased steps occurred at t=509.09090909090907. Try solving the model up to this time only or reducing dt_max (currently, dt_max=3.5458164923888056)

Terminal outputs for SPM model:
a\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\numpy\lib\function_base.py", line 488, in asarray_chkfinite
    raise ValueError(
ValueError: array must not contain infs or NaNs

**Comment by brosaplanella** [comment, marked as answer]
Hi @shant4! Note that the Chen2020 cell is an energy cell so in practice the maximum C-rate recommended by the manufacturer is 1.5C. From the modelling point of view, at high C-rates you will encounter two main issues:
1. The discharge time is very short, and you will need to provide a shorter `t_eval` and shorter `dt_max`, otherwise the solver won't converge.
2. The electrolyte will deplete. For the DFN, this means it will stop when the electrolyte gets close to zero (hence it will stop before reaching the 2.5V cut-off). For the SPM, the simulations should run (once the `dt_max` is reduced) but the results will make no physical sense as they do not include any electrolyte effect.

**Reply by Thiyaga2025** [reply]
@brosaplanella , thanks for the response, i do face similar issue at 15 A current discharge ( 3C-rate) for LGM50 cell. my solver settings with DFN model are 
solver = pybamm.CasadiSolver(atol=1e-3, rtol=1e-3, mode="safe"). the error message is "At t = 282.847, repeated recoverable residual errors". but during testing the cell was discharged at 15A till ~1090 seconds but with default "Chen 2020" parameters my simulation fails at 580 seconds. it works good at 5 and 10 A discharge. please let me know your thoughts. can i increase electrolyte concentration.

**Reply by veera-visw** [reply]
Is there a difference in particle size/diffusion coefficient between the two paramter sets.  I am guessing this could explain why one case fails earlier than the other ? With high current, you affect the flux of Lithium on the particle surface and depending on particle size/diffusion coefficient, you can reach the limits of surface concentration at different times, and usually this is where the numerics can get tricky. Just my 2 cents..

also worth trying dt_max=1; the code will take a while but you might get better convergence.

**Reply by brosaplanella** [reply]
Note the maximum recommended discharge C-rate for the LGM50 is 1.5C, above 2C you will hit electrolyte depletion which makes the solver struggle.

Alternatively, if you can get the IDAKLUSolver to work in your machine, it is better at handling events.

**Comment by brosaplanella** [comment]
This is a very interesting question that can be useful to other people, so I will convert it to a discussion.
